packages/react/src/components/stepper/Step.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+
+import { IconCheck, IconError, IconPlaybackPause } from '../../icons';
+// import core base styles
+import 'hds-core';
+import styles from './Stepper.module.scss';
+import classNames from '../../utils/classNames';
+
+export enum StepState {
+  available,
+  completed,
+  disabled,
+  attention,
+  paused,
+}
+
+export type StepProps = React.ComponentPropsWithoutRef<'button'> & {
+  /**
+   * Data test id of step
+   */
+  dataTestId?: string;
+  /**
+   * The label of the step
+   */
+  label: string;
+  /**
+   * The language of the step
+   */
+  language?: Language;
+  /**
+   * The index of the step
+   */
+  index: number;
+  /**
+   * A callback function for custom action on step click
+   */
+  onStepClick?: (event: React.MouseEvent<HTMLButtonElement>, stepIndex: number) => void;
+  /**
+   * A function for rendering a custom aria label for step's state
+   */
+  renderCustomStateAriaLabel?: (stepIndex: number, state: StepState) => string;
+  /**
+   * A function for rendering a custom step count label
+   */
+  renderCustomStepCountLabel?: (stepIndex: number, totalNumberOfSteps: number) => string;
+  /**
+   * A boolean indicating whether step is in selected state
+   */
+  selected?: boolean;
+  /**
+   * A boolean indicating whether small variant is used
+   */
+  small?: boolean;
+  /**
+   * The state of the step
+   */
+  state: StepState;
+  /**
+   * The total number of steps
+   */
+  stepsTotal: number;
+};
+
+type Language = 'en' | 'fi' | 'sv' | string;
+
+const getStepCountLabel = (language: Language, stepIndex: number, totalNumberOfSteps: number) => {
+  return {
+    en: `Step ${stepIndex + 1}/${totalNumberOfSteps}.`,
+    fi: `Vaihe ${stepIndex + 1}/${totalNumberOfSteps}.`,
+    sv: `Steg ${stepIndex + 1}/${totalNumberOfSteps}.`,
+  }[language];
+};
+
+const states = {
+  available: {
+    fi: 'Valittavissa.',
+    en: 'Available.',
+    sv: 'Valbar.',
+  },
+  completed: {
+    fi: 'Valmis.',
+    en: 'Completed.',
+    sv: 'Komplett.',
+  },
+  attention: {
+    fi: 'Vaatii huomiota.',
+    en: 'Needs attention.',
+    sv: 'Behöver uppmärksamhet.',
+  },
+  paused: {
+    fi: 'Keskeytetty.',
+    en: 'Paused.',
+    sv: 'Pausad.',
+  },
+};
+
+const getStepState = (language: Language, state: StepProps['state']) => {
+  if (state === StepState.disabled) {
+    // Button has disabled attribute, which is enough for a11y
+    return '';
+  }
+  return states[StepState[state]][language];
+};
+
+export const Step = React.forwardRef<HTMLButtonElement, StepProps>(
+  (
+    {
+      label,
+      language = 'fi',
+      index,
+      renderCustomStepCountLabel,
+      small = false,
+      state,
+      selected,
+      stepsTotal,
+      renderCustomStateAriaLabel,
+      onStepClick,
+      dataTestId,
+      ...rest
+    }: StepProps,
+    ref?: React.RefObject<HTMLButtonElement>,
+  ) => {
+    const composeAriaLabel = () => {
```

Comment:
Would it be simpler to pass the label for the step without calculating it on every step?
There is a lot of info that the step does not need necessarily to know